### PR TITLE
New ensemble new learning

### PR DESCRIPTION
**Pull request by neworderofjamie**
No filtered activity-based operation, but otherwise works - Re-adds support for Voja and PES to new-ensemble

### CODE REVIEW

**Comment by neworderofjamie** [comment]
I think this has reached the point of review! I may try and add some more tests, but I think the coverage of the learning stuff is reasonable

**Comment by mundya** [comment]
Dare I ask what you'd like to do with this history?

**Comment by studywolf** [comment]
So i'm playing around with this code now, and I tried to make two connections into `conn.learning_rule`

``` python
nengo.Connection(target, conn.learning_rule, transform=-1)
nengo.Connection(current, conn.learning_rule)
```

and it looked like the first connection was being bumped off by the second connection. When I combined the two in a separate ensemble beforehand and connected that output to the learning rule things worked fine.

**Comment by neworderofjamie** [comment]
I am pretty certain I never thought that was even possible :) I THINK it
would be relatively easy to implement. @mundy - What are your plans for the
integration of this, now worryingly out of date, branch and new_ensemble in
general into master?

**Comment by tcstewar** [comment]
> I am pretty certain I never thought that was even possible :)

:)  Yeah, all our examples only use one input to a learning_rule, but as I've been playing with learning rules more and more I keep doing this multiple connection thing.  Basically since the error population is almost always doing some sum of two different inputs, you can almost always just get rid of the error population and do two different connections into the learning rule.  So we've started getting into the habit of doing that, just for efficiency reasons....  But it is definitely quite surprising.

**Comment by neworderofjamie** [comment]
If you could raise an issue and assign it to me, I will add it at some
point in the near future. What are you guys using Nengo SpiNNaker learning
for?

**Comment by studywolf** [comment]
Robots! :smile:

**Comment by studywolf** [comment]
issue made but i am unable to assign!

**Comment by mundya** [comment]
> What are your plans for the integration of this, now worryingly out of date, branch and new_ensemble in general into master?

Unsure. Given how far ~~we've~~ I've let this slip it might be easier to do the merge in person when I'm next in Manchester (like w/b 11th April)?

**Comment by mundya** [comment]
@neworderofjamie I'll try to review this tomorrow, hopefully the merge shouldn't be too horrible... I don't think master has changed _that_ much.

**Comment by neworderofjamie** [comment]
Awesome, has new_ensemble even made it to master? :)

**Comment by mundya** [comment]
> Awesome, has new_ensemble even made it to master? :)

Yes, I think it's been in master for a bit now

File: nengo_spinnaker/builder/ensemble.py [hunk 1]
```diff
@@ -221,3 +284,31 @@ def build_neurons_probe(model, probe):
...
+    learnt_connection = probe.target.connection
+
+    # If this learning rule supports this type of probe
+    if probe.attr in learning_rule_type.probeable:
+        # If we're probing scaled encoders
+        if probe.attr == "scaled_encoders":
+            # If learning rule modifies encoders
+            if learning_rule_type.modifies == "encoders":
+                # If post-synaptic object is an ensemble, add probe
+                # to the bject operators lists of local probes
```
Comment (mundya) [inline nengo_spinnaker/builder/ensemble.py:331]: "add probe to the operator's list of local probes"

**Comment by mundya** [comment]
Manually merged into master.

**Pull request closed without merging**